internal/service/synthetics/canary.go

```diff
@@ -703,10 +714,18 @@ func expandCanaryRunConfig(l []interface{}) *synthetics.CanaryRunConfigInput {
 		codeConfig.ActiveTracing = aws.Bool(v)
 	}
 
+	if vars, ok := m["environment_variables"].(map[string]interface{}); ok {
+		ev := make(map[string]string)
+		for k, v := range vars {
+			ev[k] = v.(string)
+		}
+		codeConfig.EnvironmentVariables = aws.StringMap(ev)
+	}
```

Comment:
we can use the `flex.ExpandStringMap` method here instead e.g.
```suggestion
	if vars, ok := m["environment_variables"].(map[string]interface{}); ok && len(vars) > 0 {
		codeConfig.EnvironmentVariables = flex.ExpandStringMap(ev)
	}
```